internal/scylla/client.go

```diff
@@ -136,11 +136,65 @@ func (c *Client) doHttpRequest(req *http.Request) (resp *http.Response, temporar
 		return
 	}
 
-	temporaryErr = resp.StatusCode == http.StatusBadGateway || resp.StatusCode == http.StatusGatewayTimeout || resp.StatusCode == http.StatusTooManyRequests
+	body, err := getBody(resp)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to get response body: %w", err)
+	}
+
+	body2, err := getBody(resp)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to get response body: %w", err)
+	}
+
+	code := getErrorCode(body)
+
+	tflog.Trace(req.Context(), "getErrorCode: "+code, map[string]interface{}{
+		"code":   resp.StatusCode,
+		"status": resp.Status,
+		"body1":  string(body),
+		"body2":  string(body2),
+	})
+
+	if code == "000001" {
+		return resp, true, fmt.Errorf("too many request response")
+	}
+	temporaryErr = temporaryErr || resp.StatusCode == http.StatusBadGateway || resp.StatusCode == http.StatusGatewayTimeout || resp.StatusCode == http.StatusTooManyRequests
 	return
 }
 
+func getBody(resp *http.Response) ([]byte, error) {
+	if resp.Body == nil || resp.Body == http.NoBody {
+		return nil, nil
+	}
+	defer resp.Body.Close()
+	var buf bytes.Buffer
+	if _, err := buf.ReadFrom(resp.Body); err != nil {
+		return nil, err
+	}
+	resp.Body = io.NopCloser(bytes.NewReader(buf.Bytes()))
```

Comment:
Since I don't ammend it does not matter, since `GetBody` is going to return same body

Reply:
> GetBody is going to return same value

Then why you read into buffer instead of using `resp.GetBody`?

```
resp.Body, err = resp.GetBody()
```

Reply:
`http.Response` does not have `GetBody` method